plugin/restore/mysql/mysql.go

```diff
@@ -349,7 +349,14 @@ func (r *Restore) SwapPITRDatabase(ctx context.Context, database string, suffixT
 		return pitrDatabaseName, pitrOldDatabase, err
 	}
 
-	if _, err := db.ExecContext(ctx, "SET sql_log_bin=OFF"); err != nil {
+	// We use conn to ensure that context operations are performed within a connection(i.e. session in MySQL).
+	conn, err := db.Conn(ctx)
+	if err != nil {
+		return pitrDatabaseName, pitrDatabaseName, err
+	}
+	defer conn.Close()
+
+	if _, err := conn.ExecContext(ctx, "SET sql_log_bin=OFF"); err != nil {
```

Comment:
```suggestion
	// Set OFF the session variable sql_log_bin so that the writes in the following SQL statements will not be recorded in the binlog.
	// This is important to make the binlog replay operation idempotent.
	if _, err := conn.ExecContext(ctx, "SET sql_log_bin=OFF"); err != nil {
```